scripts/preconversion_assessment_script.py

```diff
@@ -197,27 +258,113 @@ def cleanup(required_files=None):
         )
 
 
-def verify_required_files_are_present(required_files=None):
-    if required_files is None:
-        required_files = REQUIRED_FILES
-
+def verify_required_files_are_present(required_files):
+    """Verify if the required files are already present on the system."""
+    print("Checking if required files are present on the system.")
     for required_file in required_files:
         # Avoid race conditions
         try:
+            print("Checking for file %s" % required_file.path)
             with open(required_file.path, mode="r") as handler:
                 required_file.sha512_on_system = hashlib.sha512(handler.read())
                 required_file.is_file_present = True
         except (IOError, OSError):
             required_file.is_file_present = False
 
 
+def _generate_message_key(message, action_id):
+    """
+    Helper method to generate a key field in the message composed by action_id
+    and message_id.
+    """
+    new_message = copy.deepcopy(message)
+
+    new_message["key"] = "%s::%s" % (action_id, message["id"])
+    del new_message["id"]
+
+    return new_message
+
+
+def _generate_detail_block(message):
+    """
+    Helper method to generate the detail key that is composed by the
+    remediation and diagnosis fields.
+    """
+    new_message = copy.deepcopy(message)
+    detail_block = {
+        "remediation": {"context": ""},
+        "diagnosis": {"context": ""},
+    }
+
+    detail_block["remediation"]["context"] = new_message["remediation"]
+    detail_block["diagnosis"]["context"] = new_message["diagnosis"]
+
+    del new_message["remediation"]
+    del new_message["diagnosis"]
```

Comment:
the properties always exist in original data? I mean .. are we safe from key errors here? maybe pop with default values would be safer

Reply:
That's a very good point. I have changed it to be `new_message.pop("remediation", "")` and `new_message.pop("diagnosis", "")`.